### DISCUSSION TITLE
Q: Multi-Agent Tasks/Environments supported?

### DISCUSSION THREAD

**Discussion by ConstantinRuhdorfer** [corl-team/xland-minigrid#18] [Q&A] [answered]
Hi there!

Thanks for the implementation!
We are looking into the implementation and were wondering whether multi-agent training is supported?
In the original XLand - in my understanding - some tasks are single-agent and some are multi-agent and I remember that this work used to have a disclaimer that only single agent worlds/tasks were supported. 
I saw nothing in the examples - is this still the case?

Thanks!

**Comment by Howuhh** [comment, marked as answer]
@ConstantinRuhdorfer Hi! Yes, currently only single-agent tasks are supported as they are quite easier to implement. However, I believe that multi-agent version of XLand-MiniGrid is possible (with some currently unknown runtime overhead), but I do not have plans to implement it myself in near future (2-3 month at least), I am more focused on offline meta-rl right now. 

Although, I would happily welcome any contributions from the outside regarding multi-agent envs. This will require vectorizing state and actions across multiple agents. I think rules and goals will be unchanged (only agent focused will require changes).

**Reply by ConstantinRuhdorfer** [reply]
Hi @Howuhh , 

Thanks for getting back so quickly! We might be willing to work on this since we are interested in the MARL/cooperation side of XLand but I have to think about it for a bit. Are you open to answering some questions every so often if they would emerge?

Cheers!

**Reply by Howuhh** [reply]
@ConstantinRuhdorfer Hi! Yup, I'll probably be able to answer most of the questions, but my workload for the next month will grow closer to deadlines on NeurIPS. After that I will have more free time again.

**Reply by Howuhh** [reply]
@ConstantinRuhdorfer also, do you know about https://github.com/FLAIROx/JaxMARL? Maybe it will be sufficient for your research?

**Reply by ConstantinRuhdorfer** [reply]
@Howuhh , Ah yes of course, same here. Yes, I really enjoy JaxMARL and was thinking about exploring a multi-agent  xland-minigrid for a project I have in mind. That would definitely make use of the baselines in JaxMARL but the environments there are sadly not sufficient for the project. Thank you though!
